apps/dashboard/src/app/checkout/components/client/CheckoutLinkForm.client.tsx

```diff
@@ -0,0 +1,179 @@
+      }
+
+      // Validate addresses
+      if (!checksumAddress(recipientAddress)) {
+        throw new Error("Invalid recipient address");
+      }
+      if (!checksumAddress(tokenAddress)) {
+        throw new Error("Invalid token address");
+      }
+
+      // Get token decimals
+      const tokenContract = getContract({
+        client,
+        // eslint-disable-next-line no-restricted-syntax
+        chain: defineChain(chainId),
+        address: tokenAddress,
+      });
+      const { decimals } = await getCurrencyMetadata({
+        contract: tokenContract,
+      });
+
+      // Convert amount to wei
+      const amountInWei = BigInt(Number.parseFloat(amount) * 10 ** decimals);
+
+      // Build checkout URL
+      const params = new URLSearchParams({
+        chainId: chainId.toString(),
+        recipientAddress,
+        tokenAddress,
+        amount: amountInWei.toString(),
+      });
+
+      const checkoutUrl = `${window.location.origin}/checkout?${params.toString()}`;
+
+      // Copy to clipboard
+      await navigator.clipboard.writeText(checkoutUrl);
+
+      // Show success toast
+      toast.success("Checkout link copied to clipboard.");
+    } catch (err) {
+      setError(err instanceof Error ? err.message : "An error occurred");
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  return (
+    <Card className="mx-auto w-full max-w-[500px]">
+      <CardHeader>
+        <div className="flex flex-col items-center gap-2 sm:flex-row sm:gap-2">
+          <div className="rounded-lg border border-muted p-1.5 sm:p-2">
+            <CreditCardIcon className="size-5 sm:size-6" />
+          </div>
+          <CardTitle className="text-center sm:text-left">
+            Create a Checkout Link
+          </CardTitle>
+        </div>
+      </CardHeader>
+      <CardContent>
+        <form onSubmit={handleSubmit} className="space-y-6">
+          <div className="space-y-2">
+            <label htmlFor="network" className="font-medium text-sm">
+              Network
+            </label>
+            <SingleNetworkSelector
+              chainId={chainId}
+              onChange={setChainId}
+              client={client}
+              className="w-full"
+            />
+          </div>
+
+          <div className="space-y-2">
+            <label htmlFor="recipient" className="font-medium text-sm">
+              Recipient Address
+            </label>
+            <Input
+              id="recipient"
+              value={recipientAddress}
+              onChange={(e) => setRecipientAddress(e.target.value)}
+              placeholder="0x..."
+              required
+              className="w-full"
+            />
+          </div>
+
+          <div className="space-y-2">
+            <label htmlFor="token" className="font-medium text-sm">
+              Token Address
+            </label>
+            <Input
+              id="token"
+              value={tokenAddress}
```

Comment:
tough to make ppl enter the native token address, really needs to be a dropdown with a few popular options + custom

selecting custom then shows the address field